packages/webapp/src/containers/Finances/CustomRevenueType/CustomRevenueRadios.jsx

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { Main, Info } from '../../../components/Typography';
+import RadioGroup from '../../../components/Form/RadioGroup';
+import { useTranslation } from 'react-i18next';
+import { AGRICULTURE_ASSOCIATED, CROP_GENERATED } from './constants';
+import PropTypes from 'prop-types';
+
+/**
+ * Radio groups for setting and viewing the custom revenue type properties
+ *
+ * @param {Object} control - control object from React Hook Form
+ * @param {function} watch - watch method from React Hook Form
+ * @param {string} view - The view mode ('read-only', 'add' or 'edit')
+ * @returns {JSX.Element}
+ */
+
+function CustomRevenueRadios({ control, watch, view }) {
+  const { t } = useTranslation();
+
+  const isAgricultureAssociated = watch(AGRICULTURE_ASSOCIATED);
+
+  return (
+    <>
+      <Main style={{ paddingBlock: '12px' }}>
+        {t('REVENUE.ADD_REVENUE.ASSOCIATED_WITH_AGRICULTURE')}
+      </Main>
+
+      <RadioGroup
+        hookFormControl={control}
+        name={AGRICULTURE_ASSOCIATED}
+        radios={[
+          {
+            label: t('common:YES'),
+            value: true,
+          },
+          { label: t('common:NO'), value: false },
+        ]}
+        required
+        disabled={view === 'edit' || view === 'read-only'}
+      />
+      {view === 'add' && <Info>{t('REVENUE.ADD_REVENUE.CANNOT_BE_CHANGED')}</Info>}
```

Comment:
Should we show this information on the Edit view too so that it's clearer to the user why it's disabled and can't be edited?

Reply:
Yes, I like that idea! The given wording only makes sense for add, though, I think:

![Screenshot 2023-10-12 at 9 58 04 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/cbb5e1a8-4cfb-4395-9af2-6d4e072dd531)

For edit, do you think just "This cannot be changed" is enough? Or "This cannot be changed once set?"

Reply:
"This cannot be changed" makes sense to me!